src/ucp/proto/proto_single.c

```diff
@@ -29,8 +29,15 @@ ucp_proto_single_init_priv(const ucp_proto_single_init_params_t *params,
     ucp_lane_index_t lane;
     ucs_status_t status;
 
+    /* Slow path lanes are the last 3 (UCX_MAX_SLOW_PATH_LANES) lanes,
+     * so we need to shift them by 5 (UCX_MAX_FAST_PATH_LANES) */
+    ucp_lane_map_t slow_path_lanes = UCS_MASK(UCP_MAX_SLOW_PATH_LANES)
+                                     << UCP_MAX_FAST_PATH_LANES;
+
     num_lanes = ucp_proto_common_find_lanes(&params->super, params->lane_type,
-                                            params->tl_cap_flags, 1, 0, &lane);
+                                            params->tl_cap_flags, 1,
+                                            /* Exclude slow path lanes */
+                                            slow_path_lanes, &lane);
```

Comment:
params->lane_type should be such that only fast path will be returned
we can assert that ucp_proto_common_find_lanes really returns fast path lane

Reply:
won't be easier to check params->lane_type?

Reply:
or did you mean check we don't exceed MAX_FAST_PATH_LANES?

Reply:
we don't need to add logic to force fast paths lanes here
can just assert the returned lane is < MAX_FAST_PATH_LANES

Reply:
if we assert it, it means that lane must be fast, from function call it's not clear
maybe add an assertion that `params->lane_type` is also fast?